### PR TITLE
support `@base/math` `@pal/screen-adapter` and `@pal/minigame` module

### PR DESCRIPTION
**Pull request by PPpro**
Re: 
https://github.com/cocos/cocos-engine/issues/14293
https://github.com/cocos/cocos-engine/issues/16363
https://github.com/cocos/cocos-engine/issues/16419

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/renderer/render-data.ts [hunk 1]
```diff
@@ -26,10 +26,11 @@ import { DEBUG, JSB } from 'internal:constants';
 import { assert } from '@base/debug';
 import { assertIsTrue } from '@base/debug/internal';
 import { memop } from '@base/utils';
+import { Color } from '@base/math';
```
Comment (minggo) [inline cocos/2d/renderer/render-data.ts:29]: Why Color belongs to math module?

File: cocos/core/platform/sys.ts [hunk 2]
```diff
@@ -24,12 +24,11 @@
 */
 
 import { systemInfo, NetworkType, Language, OS, Platform, BrowserType, Feature } from '@pal/system-info';
-import { screenAdapter } from 'pal/screen-adapter';
+import { screenAdapter } from '@pal/screen-adapter';
```
Comment (minggo) [inline cocos/core/platform/sys.ts:27]: screen-adapter is as separate module?

Comment (PPpro) [reply cocos/core/platform/sys.ts:27]: yes, it is

File: cocos/2d/renderer/render-data.ts (hunk 1 above)
Comment (PPpro) [reply cocos/2d/renderer/render-data.ts:29]: for our current implementation, it does belong to math module, and it's our public interface design

Comment (PPpro) [reply cocos/2d/renderer/render-data.ts:29]: we could migrate `Color` to `core` layer, but it should follow the interface deprecation workflow, to deprecate the public interface `math.Color`

and that is not what this PR would do

**Review by minggo — APPROVED** [review]

**Pull request merged**